esti/system_test.go

```diff
@@ -73,11 +73,9 @@ func setupTest(t *testing.T) (context.Context, logging.Logger, string) {
 	return ctx, logger, repo
 }
 
-func tearDownTest(t *testing.T, repoName string) {
+func tearDownTest(repoName string) {
 	ctx := context.Background()
-	name := makeRepositoryName(t.Name())
-	logger := logger.WithField("testName", name)
-	deleteRepository(ctx, t, repoName)
+	deleteRepositoryIfAskedTo(ctx, repoName)
```

Comment:
Not a blocker, but I recommend the env var to be for teardown which will be checked both in tearDownTest and in deleteRepositoryIfAskedTo

Reply:
why double checking?

Reply:
Simply because in the future we might want to add more logic to the teardown function and this flag will be relevant for it. Optimally the check for this flag should be only on this function. But since you explicitly use the delete repository function we need to add the check also there.
But again, not blocking